Explorer/Assets/DCL/WebRequests/WebRequestController.cs

```diff
@@ -14,22 +18,81 @@ public class WebRequestController : IWebRequestController
         private readonly IWebRequestsAnalyticsContainer analyticsContainer;
         private readonly IWeb3IdentityCache web3IdentityCache;
         private readonly IRequestHub requestHub;
+        private readonly string csvPath;
+        private static readonly object csvLock = new object();
 
-        public WebRequestController(IWebRequestsAnalyticsContainer analyticsContainer, IWeb3IdentityCache web3IdentityCache, IRequestHub requestHub)
+        public WebRequestController(
+            IWebRequestsAnalyticsContainer analyticsContainer,
+            IWeb3IdentityCache web3IdentityCache,
+            IRequestHub requestHub)
         {
             this.analyticsContainer = analyticsContainer;
             this.web3IdentityCache = web3IdentityCache;
             this.requestHub = requestHub;
+
+            // Set up CSV file path in persistent data path
+            csvPath = Path.Combine(Application.persistentDataPath, "request_metrics.csv");
+
+            // Create CSV header if file doesn't exist
+            if (!File.Exists(csvPath))
+            {
+                lock (csvLock)
+                {
+                    if (!File.Exists(csvPath))
+                    {
+                        File.WriteAllText(csvPath,
+                            "Timestamp,RequestType,URL,RequestTime,OperationTime,TotalTime,Status,AttemptNumber,ErrorMessage\n");

```

Comment:
Instead, open the file, and keep it open. If you want to ensure writes reach disk, flush. You remember how we talked that performance is important? This is it. Time spent re-opening files on the main thread is time taken away from pushing draw calls.

Reply:
I agree, this wasn't going to be merged! I am just testing the build.